src/Compilers/Core/Portable/DocumentationCommentId.cs

```diff
@@ -344,21 +374,59 @@ public override void VisitMethod(IMethodSymbol symbol)
             public override void VisitNamespace(INamespaceSymbol symbol)
             {
                 _builder.Append("N:");
-                _generator.Visit(symbol);
+                AppendNamespace(symbol, _builder);
             }
 
             public override void VisitNamedType(INamedTypeSymbol symbol)
             {
                 _builder.Append("T:");
-                _generator.Visit(symbol);
+                AppendNamedType(symbol, _builder);
             }
 
-            private class Generator : SymbolVisitor<bool>
+            private static void AppendNamespace(INamespaceSymbol symbol, StringBuilder builder)
+            {
+                if (symbol.IsGlobalNamespace)
+                    return;
+
+                if (symbol.ContainingNamespace is { IsGlobalNamespace: false })
+                {
+                    AppendNamespace(symbol.ContainingNamespace, builder);
+                    builder.Append('.');
+                }
+
+                builder.Append(EncodeName(symbol.Name));
+            }
+
+            private static void AppendNamedType(INamedTypeSymbol symbol, StringBuilder builder)
+            {
+                if (symbol.ContainingSymbol is INamedTypeSymbol parentType)
+                {
+                    AppendNamedType(parentType, builder);
+                    builder.Append('.');
+                }
+                else if (symbol.ContainingSymbol is INamespaceSymbol { IsGlobalNamespace: false } parentNamespace)
+                {
+                    AppendNamespace(parentNamespace, builder);
+                    builder.Append('.');
+                }
+
+                builder.Append(EncodeName(symbol.Name));
+
+                if (symbol.TypeParameters.Length > 0)
+                {
+                    builder.Append('`');
+                    builder.Append(symbol.TypeParameters.Length);
+                }
+            }
+
+            private sealed class DeclarationGenerator : SymbolVisitor
```

Comment:
> private sealed class DeclarationGenerator : SymbolVisitor [](http://example.com/codeflow?start=11&length=58)

In my opinion the changes from this point to the end of the file that are not related to the introduction of the `Failed` flag in this and the enclosing type are unrelated to the stated goal of the PR. I think they should be reverted. Perhaps with exception of the type rename.

Reply:
wfm.